packages/api/src/external/commonwell/patient-external-data.ts

```diff
@@ -97,3 +98,56 @@ export const setCommonwellId = async ({
     return updatedPatient.update({ data: updatedData }, { transaction });
   });
 };
+
+export async function updatePatientScheduledQueryRequestId({
+  patient,
+}: {
+  patient: Patient;
+}): Promise<void> {
+  const patientFilter = {
+    id: patient.id,
+    cxId: patient.cxId,
+  };
+
+  const scheduledDocQueryRequestId = getCWData(
+    patient.data.externalData
+  )?.scheduledDocQueryRequestId;
+
+  if (!scheduledDocQueryRequestId) return;
+
+  await executeOnDBTx(PatientModel.prototype, async transaction => {
+    const existingPatient = await getPatientOrFail({
+      ...patientFilter,
+      lock: true,
+      transaction,
+    });
+
+    const externalData = existingPatient.data.externalData ?? {};
+
+    const updatedExternalData = {
+      ...externalData,
+      COMMONWELL: {
+        ...externalData.COMMONWELL,
+        scheduledDocQueryRequestId: undefined,
+      },
+    };
+
+    const updatedPatient = {
+      ...existingPatient,
+      data: {
+        ...existingPatient.data,
+        externalData: updatedExternalData,
+      },
+    };
+
+    await PatientModel.update(updatedPatient, {
+      where: patientFilter,
+      transaction,
+    });
+
+    queryAndProcessDocuments({
+      patient: updatedPatient,
+      requestId: scheduledDocQueryRequestId,
+    });
```

Comment:
This seems quite important to be part of this function. Not only this function is doing more than what's supposed to, but also we have stuff inside the DB tx that we don't need to.

It would be good to follow the pattern we built for CQ, unless that's problematic - in which case it would be good to ticket a fix there.